vernac/himsg.ml

```diff
@@ -568,7 +568,9 @@ let rec explain_evar_kind env sigma evk ty = function
       str " in the partial instance " ++ pc ++
       str " found for ") ++
       explain_evar_kind env sigma evk'
-      (pr_leconstr_env env sigma ty') (snd evi.evar_source)
+        (pr_leconstr_env env sigma ty') (snd evi.evar_source)
+  | Evar_kinds.RecordFieldEvar (field, cons) ->
+     str "field `" ++ Names.Constant.print field ++ str "` of record `" ++ Libnames.pr_reference cons ++ str "`"
```

Comment:
You want to report on the type of the record. If I'm not mistaken, you are taking the name of the constructor of the record (typically `Build_foo` wrt #6394).

Apparently `sort_fields` does not have the information. So, two solutions. Maybe it is not necessary to have the type of the record and we renounce to display the information. Or we think it worth to have it, but it seems to me that `Recordops.find_projection` should be more informative and return this information. 

Assuming you'd like to go the second way, this requires a bit of extra work. My feeling is that the `projection_table` in `recordops.ml` should actually bind the projection names to the name of the record they belong to rather than directly to a `struc_typ`. This means changing the `Cmap.add proj struc` to `Cmap.add proj ind` in `Recordops.load_structure`. Then, `find_projection` would be in two steps: first find the type, and, second, find the `struc_typ` from the type in `structure_table`. Then, it would be easy to extend `Recordops.find_projection` so that it returns also the name of the record, which could then be displayed in the error message.

But the first solution would probably be an as good compromise. Do how you feel it.

Reply:
>Apparently sort_fields does not have the information. 

It does, the `record.Recordops.s_CONST` is a `Names.constructor = Names.inductive * int`.

Reply:
How stupid am I. Let me edit the message.